src/CloudLfs.Core/Brokers/BlobBroker.cs

```diff
@@ -15,26 +14,36 @@ namespace Microsoft.MixedReality.CloudLfs.Brokers
     public class BlobBroker : IBlobBroker
     {
         private Uri _storageUri;
+
         private BlobServiceClient _blobServiceClient;
 
+        private readonly BlobContainerClient containerClient;
+
         public BlobBroker(Uri storageUri)
         {
             _storageUri = storageUri;
             _blobServiceClient = new BlobServiceClient(_storageUri, new DefaultAzureCredential());
+            containerClient = _blobServiceClient.GetBlobContainerClient("objects");
+        }
+
+        public async Task<Response> DownloadAsync(string blobName, IProgress<long> progress, Stream contentStream, CancellationToken cancellationToken)
+        {
+            var blobClient = containerClient.GetBlockBlobClient(blobName);
+            return await blobClient.DownloadToAsync(contentStream, new BlobDownloadToOptions { ProgressHandler = progress } , cancellationToken);
         }
 
-        public async Task<Response> DownloadAsync(string blobName, IProgress<TransferStatus> progress, Stream contentStream, CancellationToken cancellationToken)
+        public async Task<Response> DownloadAsync(string blobName, IProgress<long> progress, Stream contentStream, long startBytes, long endBytes, CancellationToken cancellationToken)
         {
-            var containerClient = _blobServiceClient.GetBlobContainerClient("objects");
-            BlobClient blobClient = containerClient.GetBlobClient(blobName);
-            return await blobClient.DownloadToAsync(contentStream, cancellationToken);
+            var blobClient = containerClient.GetBlockBlobClient(blobName);
+            var result = await blobClient.DownloadStreamingAsync(new HttpRange(startBytes, (endBytes - endBytes)), progressHandler: progress, cancellationToken: cancellationToken);

```

Comment:
why `endBytes - endBytes`?

Reply:
actually, should we remove this method? I'm guessing this was testing the chunk downloading